odoo/addons/base/models/ir_http.py

```diff
@@ -174,86 +158,31 @@ def _serve_attachment(cls):
             return response
 
     @classmethod
-    def _serve_fallback(cls, exception):
+    def _serve_fallback(cls):
         # serve attachment
         attach = cls._serve_attachment()
         if attach:
             return attach
-        return False
```

Comment:
TL;DR, `return False` outside of public model methods is a cargo cult in Odoo. This method *optionally* returns a response, thus should return `None` when no response could be served.

Within Odoo public models method have to return `False` because historically many XML-RPC lacked a way to serialize `None`. The closest thing to `None` that is XML-RPC serializable is `False` thus it was decided to `return False` when a method could be called via XML-RPC.

The result of this **hack** is that most Odoo employees think that we cannot `return None`, although it is a perfectly valid return value with regular function and private model methods.

If we were to add type-hints to the function, we would document that the return type is `Optional[odoo.http.Response]`. We could argue that it should raises a `werkzeug.exceptions.NotFound` exception instead of optionally returning a response (just like the other `serve_x` method in `http.py`) but that's another story.

With `return False`, the signature becomes `-> Union[odoo.http.Response, bool]`. The function returns a Response or a boolean. One would assume that the literal `True` could be returned but it is not the case.

None expresses "the lack of something". `False` expresses "the opposite of True`. In this method, when we don't return a response, we *lack a response to return`. If the implementation was sending the http response on the network socket itself instead of merely returning it then we could return `False` when no response was delivered because the opposite was true, `True` would have been returned to express that a response was delivered.